### PR TITLE
Add versioning strategy discussion

### PR DESCRIPTION
**Pull request by kylegach**
### Description
Documenting our discussion around versioning.

### Motivation and context

For posterity.

### Types of changes

- Other (provide details below)

Docs

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support) [n/a]
* [x] Automated tests written and passing [n/a]
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered [n/a]
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered [n/a]
* [x] Documentation created or updated [n/a]

### How does this PR make you feel?

Decide

### CODE REVIEW

**Comment by brentertz** [comment]
Whichever way we go, it will need documented for consumers somewhere more visibly than this document, e.g. in the readme and in the site.

**Comment by kylegach** [comment]
Agreed 💯%.

**Comment by brentertz** [comment]
Strategy B sounds like a 👑  PITA, particularly dealing with separating commits and the current release:manual vs. semantic process.

**Comment by kylegach** [comment]
Agreed there, too. My suggestion is the third strategy: a new npm org.

File: docs/discussions/versioning.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+   | ComponentUtils | 1.0.0 |
+
+   Then we release Table, which we don't believe to be stable yet...
+
+   | Component | Version |
+   | :-------- | :------ |
+   | Button | 1.0.0 |
+   | Card | 1.0.0 |
+   | ComponentUtils | 1.0.0 |
+   | Table | 0.0.0 |
```
Comment (brentertz) [inline docs/discussions/versioning.md]: You can't actually release as `0.0.0`.  `0.1.0` is probably more accurate.

**Comment by brentertz** [comment]
Your descriptions of how things play out down the road sound like a challenging maintenance burden.  I do not currently believe that our component versions need to be kept in sync and I believe that it would be quite difficult given our current architecture.

If you want all component versions to be in sync, then just change the lerna versioning mode to "Fixed/Locked".  Lerna's own docs however suggests "Independent mode ... makes sense for a group of components."  

What I am hearing is the desire to version the library, which is valid, though I think it is dangerous to attempt to predict a strategy for 2.0 based on design changes.  Following a 1.0 release, semver would need to win, regardless.  As our components are individually consumable, and we do not have an aggregate package, this is not entirely possible at present.  Perhaps we can solve it from a different angle.  Once we have an aggregate package, consumers that choose to use it won't ever have to deal with more than one version.  In the interim, peer dependencies within each package contain the associated supported versions.  Keeping it simple, consumers should always attempt to use the latest version of all mineral components.

If this is truly this important, we'll need to reconsider the strategy of publishing independent components and therefore also using lerna at all.  This wouldn't be the end of the world, but it would take some rework.

As the 1.0 ship has sailed, due to being honest about changes and using semantic versioning from day 1, I think that the only way to reset the baseline is to move to a new org.

**Comment by brentertz** [comment]
@kylegach Do we need this PR anymore?  It has been resolved by switching to a single library package model rather than publishing independently packaged/versioned components.  This allows us to reset versioning at 0 and stay below 1.0.0 until we deem appropriate.  Documentation has been updated to include our new versioning strategy.

**Pull request closed without merging**